gapic/schema/wrappers.py

```diff
@@ -1845,13 +1852,32 @@ def paged_result_field(self) -> Optional[Field]:
             if not field or field.type != source_type:
                 return None
 
-        # The request must have max_results or page_size
+        # The request must have page_size (or max_results if legacy API)
         page_fields = (
             self.input.fields.get("max_results", None),
             self.input.fields.get("page_size", None),
         )
         page_field_size = next((field for field in page_fields if field), None)
-        if not page_field_size or page_field_size.type != int:
+        if not page_field_size:
+            return None
+
+        # If the a legacy API uses the UInt32Value and Int32Value wrappers,
+        # the service package must be in the allowlist.
+        package_name = self.input.meta.address.proto_package
+
+        if page_field_size.type == int or (
+            # The following additional checks are related to several members of the BQ family of
+            # APIs, which use the legacy wrapper types: "UInt32Value" and "Int32Value"}
+            # for max_results.
+            # NOTE:
+            #   bigquery_v2 should be paginated
+            #   but bigquery_connection_v1beta1 should NOT be paginated
+            isinstance(page_field_size.type, MessageType)
+            and page_field_size.type.message_pb.name in {"UInt32Value", "Int32Value"}
+            and package_name in self.PACKAGES_WITH_ALLOWED_WRAPPERS
```

Comment:
Please remove this line. We can add it back if we find that this is going to break other APIs.
```suggestion
```